mne/stats/cluster_level.py

```diff
@@ -582,16 +576,18 @@ def _do_permutations(X_full, slices, threshold, tail, connectivity, stat_fun,
         perm_clusters_sums = out[1]
 
         if len(perm_clusters_sums) > 0:
-            max_cluster_sums[seed_idx] = np.max(perm_clusters_sums)
+            max_cluster_sums[oi] = np.max(perm_clusters_sums)
         else:
-            max_cluster_sums[seed_idx] = 0
+            max_cluster_sums[oi] = 0
+
+        pb[pb_idx[oi]] = True
 
     return max_cluster_sums
 
 
 def _do_1samp_permutations(X, slices, threshold, tail, connectivity, stat_fun,
                            max_step, include, partitions, t_power, orders,
-                           sample_shape, buffer_size, progress_bar):
+                           sample_shape, buffer_size, pb, pb_idx):
```

Comment:
why does it become more complicated than doing progress_bar.update()? can the .update() work in parallel?

Reply:
Yes it's slightly more complicated, but now the `.update()` works properly in parallel by using a NumPy memmapped array. It fixes the problems on `master` that we have sometimes with progress jumping forward and backward.